prac_08/taxi_simulator.py

```diff
@@ -0,0 +1,77 @@
+"""
+CP1404 - Practical 08
+Taxi Simulator
+"""
+
+from prac_08.taxi import Taxi
+from prac_08.silver_service_taxi import SilverServiceTaxi
+
+MENU = "q)uit, c)hoose taxi, d)rive"
+
+
+def main():
+    """Taxi simulator program."""
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    bill = 0
+
+    print("Let's drive!")
+    print(MENU)
+    choice = input(">>> ").lower()
+    while choice != "q":
+        if choice == "c":
+            print("Taxis available:")
+            print_taxis(taxis)
+            chosen_taxi = get_taxi(taxis)
+        elif choice == "d":
+            if chosen_taxi:
+                drive_distance = get_valid_distance()
+                chosen_taxi.drive(drive_distance)
```

Comment:
Just realised I looked at the out of date version first, whoops. Good job changing it to store the (Taxi) object instead of the position though 👍